src/Storages/StorageS3.cpp

```diff
@@ -615,6 +615,11 @@ std::shared_ptr<StorageS3Source::IteratorWrapper> StorageS3::createFileIterator(
     }
 }
 
+bool StorageS3::isColumnOriented() const
+{
+    return format_name != "Distributed" && FormatFactory::instance().checkIfFormatIsColumnOriented(format_name);
```

Comment:
I already don't remember what is `Distributed` format. Maybe we should prohibit it from using with anything expect `StorageFile`.